crates/resolver-tests/src/lib.rs

```diff
@@ -884,3 +1179,121 @@ pub fn assert_same<A: PartialEq + std::fmt::Debug>(a: &[A], b: &[A]) {
     assert_eq!(a.len(), b.len(), "not equal\n{a:?}\n{b:?}");
     assert_contains(b, a);
 }
+
+#[cfg(test)]
+mod sat_test {
+    use super::*;
+
+    #[test]
+    fn test_features() {
+        let reg = registry(vec![
+            pkg!("a"),
+            pkg!("b"),
+            pkg_dep_with(
+                "image",
+                vec!["a".to_opt_dep(), "b".to_opt_dep(), "jpg".to_dep()],
+                &[("default", &["a"]), ("b", &["dep:b"])],
+            ),
+            pkg!("jpg"),
+            pkg!("log"),
+            pkg!("man"),
+            pkg_dep_with("rgb", vec!["man".to_opt_dep()], &[("man", &["dep:man"])]),
+        ]);
+
+        let root_summary = pkg_dep_with(
+            "root",
+            vec![
+                "image".to_dep_with(&["b"]),
+                "log".to_opt_dep(),
+                "rgb".to_opt_dep(),
+            ],
+            &[
+                ("default", &["log", "image/default"]),
+                ("man", &["rgb?/man"]),
+            ],
+        );
+
+        let checks = [
+            (
+                &CliFeatures::new_all(true),
+                r#"used:
```

Comment:
Do we want to be testing that each of these matches a hardcoded value? 
- If no: If we move them to `crates/resolver-tests/tests/resolve.rs`, we could use `resolve_and_validated` to make sure we get the same output as cargo, without needing to maintain the hardcoded value.
- If yes: The rest of cargo is moving toward using `snapbox` for maintaining these hardcoded strings.